src/migrate.js

```diff
@@ -40,20 +41,22 @@ if ( jQuery.migrateTrace === undefined ) {
 // Forget any warnings we've already given; public
 jQuery.migrateReset = function() {
 	warnedAbout = {};
-	jQuery.migrateWarnings.length = 0;
+	jQuery.migrateWarnings.length = jQuery.migrateWarnings.asErrors.length = 0;
 };
 
 function migrateWarn( msg ) {
 	var console = window.console;
 	if ( !warnedAbout[ msg ] ) {
 		warnedAbout[ msg ] = true;
 		jQuery.migrateWarnings.push( msg );
+		jQuery.migrateWarnings.asErrors.push( new Error( msg ) );
 		if ( console && console.warn && !jQuery.migrateMute ) {
 			console.warn( "JQMIGRATE: " + msg );
 			if ( jQuery.migrateTrace && console.trace ) {
 				console.trace();
 			}
 		}
+		jQuery( window ).trigger( "migrateWarning" );
```

Comment:
The only time we currently call back into jQuery is to shim a method for warnings. This introduces a situation where we are using the host jQuery's functionality, which makes me nervous. It could potentially cause a recursive crash if Migrate was to give `.trigger()` warnings in the future.